src/qilib/configuration_helper/adapters/keithley_dmm6500_adapter.py

```diff
@@ -27,13 +27,16 @@
 
 
 class Keithley6500InstrumentAdapter(CommonInstrumentAdapter):
-    def __init__(self, address: str, instrument_name: Optional[str] = None) -> None:
+    def __init__(self, address: str, instrument_name: Optional[str] = None,
+                 instrument_class: Optional[Type[Keithley_6500]] = None) -> None:
         super().__init__(address, instrument_name)
         if address[0:5] == 'TCPIP':
             # Perform a http request to the instrument's IP address before opening the device to make sure any
             # stuck state is unstuck.
             self._send_request_to_instrument(address)
```

Comment:
This will fire always, right? Even in the case of the mocked instrument. Will we fix this in the `MockKeithley6500InstrumentAdapter` with again a patch?

Reply:
hmm... I guess it will fire only if the condition evaluates to True. in the case of a mocked instrument that will probably not be the case? would it? the address would be a string that won't start with TCPIP

Reply:
It could be that the address is a fake one, but I am not sure. That completely depends on how we set up the mocking of the system. To not disturb the flow of the instruments, we might want to give addresses that are as close to the originals as possible. That said, we can mock out the underlying http request in the mocked instrument adapter.

Reply:
correct, we can deal with this in the mock